### ISSUE TITLE
React 16 + shouldComponentUpdate

### ISSUE THREAD

**Issue by cambirch** [bvaughn/react-virtualized#861]
React 16 has started to warn on components that implement PureComponent and also implement their own shouldComponentUpdate method.

I am currently getting a number of warnings from react-virtualized implementing PureComponent and its own custom shouldComponentUpdate.  Currently WindowScroller, AutoSizer, List and Grid are all generating warnings.

> Warning: Grid has a method called shouldComponentUpdate(). shouldComponentUpdate should not be used when extending React.PureComponent. Please extend React.Component if shouldComponentUpdate is used.
![2017-10-31_0734](https://user-images.githubusercontent.com/2991289/32226376-13a38cf2-be0e-11e7-8e9e-d350a3e3de12.png)

I am running version 9.12.0 and quickly spot checking Grid says that it still extends PureComponent and implements its own shouldComponentUpdate() method.

Unfortunately I'm currently under a crunch so I'm going to have to ignore the warning for now, but I figured I'd be a good citizen and report it.  If its unfixed by the time I'm finished the crunch I'll hopefully be able to circle back and put together a fix (I assume the React warning is because there is no actual benefit from PureComponent if you are replacing the shouldComponentUpdate with a custom one).

**Comment by TrySound** [comment]
@cambirch  But virtualized does not use shouldComponentUpdate, show how do you use it.

**Comment by cambirch** [comment]
Your right I took a look at the source again and I was mistaking componentWillUpdate for shouldComponentUpdate.

The interesting thing is I don't directly use shouldComponentUpdate either.  But I do use MobX (which does)... strangely I have been having various non-mobx, PureComponent (mine and others) that trigger this warning.

I thought I had actually found a case.  I'm guessing that some tool I'm using must be wrapping everything in a shouldComponentUpdate for some type of tracking purposes.
